pkg/helm/actions/install_chart.go

```diff
@@ -170,17 +175,67 @@ func InstallChartAsync(ns, name, url string, vals map[string]interface{}, conf *
 		ch.Metadata.Annotations = make(map[string]string)
 	}
 	ch.Metadata.Annotations["chart_url"] = url
+	fmt.Println(ch.Metadata.Name, ch.Metadata.Version, "Here")
+	// Set up channel on which to send signal notifications.
+	// We must use a buffered channel or risk missing the signal
+	// if we're not ready to receive when the signal is sent.
 
+	ctx, cancel := context.WithCancel(context.Background())
 	cmd.Namespace = ns
+	cSignal := make(chan os.Signal, 2)
+	signal.Notify(cSignal, os.Interrupt, syscall.SIGTERM)
 	go func() {
-		_, err := cmd.Run(ch, vals)
-		if err != nil {
-			createSecret(ns, name, 1, coreClient, err)
-		}
+		<-cSignal
+		cancel()
+	}()
+	go func() {
+		cancelChan := make(chan bool, 1)
+		go func() {
+			label := fmt.Sprintf("owner=helm,name=%v,version=%v", name, 1)
+			secretList, err := coreClient.Secrets(ns).Watch(context.TODO(), metav1.ListOptions{LabelSelector: label, Watch: true})
+			if err != nil {
+				return
+			}
+			start := time.Now()
+			for {
+				event := <-secretList.ResultChan()
+				if event.Object != nil {
+					obj := event.Object.(*kv1.Secret)
+					fmt.Println("+++++++++++", obj.Labels["status"])
+					if obj.Labels["status"] == "uninstalling" {
+						fmt.Println("__Reached Here___yesssss")
+						cancelChan <- true
```

Comment:
why aren't you just calling the context cancel function?

this "cancelChan" doesn't appear to be wired to anything, at least nothing that's going to interrupt the `cmd.RunWithContext` call.

Reply:
I did try calling the cancel method directly but that did not cancel the cmd.RunWithContext.

Reply:
Calling the context method directly is invoking the ctx.Done() but resources still show up. I am looking further in https://github.com/helm/helm/blob/main/pkg/action/install.go for the reason that despite context cancellation why does other resources show up.

Reply:
canceling the context really has no effect on RunWithContext until the code gets to the point where this goroutine has been launched:
https://github.com/helm/helm/blob/main/pkg/action/install.go#L351

and even then, it just affects that goroutine, not the overall RunWithContext function.  (Nothing else in RunWithContext references the context or checks if it is canceled)

furthermore, the logic in handleContext just switches between whether it invokes reportToRun or not, based on whether the context was canceled or not:
https://github.com/helm/helm/blob/5bf273d81ba7da1816a55983fae6a6cf4ca29af2/pkg/action/install.go#L426

so i don't see any reason to expect that canceling the context would prevent resource creation.  (that is, the go routine launched from RunWithContext that is running performInstall isn't going to be interrupted by any of this, that i see)

Reply:
Context cancellation does stop the installation , for example the way it is being triggered is https://github.com/helm/helm/blob/main/cmd/helm/install.go#L272.

The  reasons why context cancellations does not work for the scenario  which i am handling  is that since i am deleting the  release , the status of the release does not get updated and hence the installation does not stop.